### PR TITLE
ACS-3353: POST support for "other fields"

### PR DESCRIPTION
**Pull request by krdabrowski**

### CODE REVIEW

**Comment by krdabrowski** [comment]
First please merge: https://github.com/Alfresco/alfresco-tas-restapi/pull/231 and remove `-SNAPSHOT` from `dependency.tas-restapi.version`

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java [hunk 1]
```diff
@@ -257,6 +271,71 @@ public void createRulesWithOneError()
...
+        rule.assertThat().field("id").isNotNull()
+            .assertThat().field("name").is(RULE_NAME_DEFAULT)
+            .assertThat().field("triggers").is(List.of("inbound"));
+    }
+
+    /** Check we can create a rule without error script. */
+    @Test (groups = { TestGroup.REST_API, TestGroup.RULES })
+    public void createRuleWithoutErrorScript()
+    {
+        RestRuleModel ruleModel = createRuleModelWithDefaultName();
```
Comment (tpage-alfresco) [inline packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java:310]: These three tests (createRuleWithoutDescription, createRuleWithoutTriggers and createRuleWithoutErrorScript) make the same API call. Maybe we could combine them into a single test called createRuleWithMinimalFields?

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/RulesTestsUtils.java [hunk 2]
```diff
@@ -29,13 +29,47 @@
...
 public class RulesTestsUtils
 {
+    static final String RULE_NAME_DEFAULT = "ruleName";
+    static final String RULE_DESCRIPTION_DEFAULT = "rule description";
+    static final boolean RULE_ENABLED_DEFAULT = true;
+    static final boolean RULE_CASCADE_DEFAULT = true;
+    static final boolean RULE_ASYNC_DEFAULT = true;
+    static final boolean RULE_SHARED_DEFAULT = false;
+    static final String RULE_ERROR_SCRIPT_DEFAULT = "error-script";
+    static final List<String> ruleTriggersDefault = List.of("inbound", "update", "outbound");
```
Comment (tpage-alfresco) [inline packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/RulesTestsUtils.java:44]: `ruleTriggersDefault` -> `RULE_TRIGGERS_DEFAULT`

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/RulesTestsUtils.java [hunk 3]
```diff
@@ -58,11 +92,19 @@ public static RestRuleModel createRuleModel(String name, List<RestActionBodyExec
...
         restActionModel.setActionDefinitionId("add-features");
         restActionModel.setParams(Map.of("aspect-name", "{http://www.alfresco.org/model/audio/1.0}audio", "actionContext", "rule"));
         return restActionModel;
     }
+
+    public static RestCompositeConditionDefinitionModel createEmptyConditionModel()
+    {
+        RestCompositeConditionDefinitionModel conditions = new RestCompositeConditionDefinitionModel();
+        conditions.setInverted(false);
+        conditions.setBooleanMode("and");
```
Comment (tpage-alfresco) [inline packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/RulesTestsUtils.java:107]: These two fields should be optional (with a default), so we should be able to leave them as `null`. If we can't do that at the moment then let's make a separate change to fix it.

File: remote-api/src/main/java/org/alfresco/rest/api/model/rules/RuleTrigger.java [hunk 4]
```diff
@@ -31,31 +31,15 @@
...
     {
-        for (RuleTrigger ruleTrigger : values())
-        {
-            if (ruleTrigger.value.equals(value)) {
-                return ruleTrigger;
-            }
-        }
-
-        return null;
+        return RuleTrigger.valueOf(value.toUpperCase());
```
Comment (tpage-alfresco) [inline remote-api/src/main/java/org/alfresco/rest/api/model/rules/RuleTrigger.java:43]: Nice simplification :+1:

**Review by tpage-alfresco — APPROVED** [review]
Looks really good.

**Pull request merged**